File: game-price-tracker-be/tests/domain/game_price.test.ts
import { Currency } from '../../src/game-price-tracker-be/domain/value_object/currency';
import { GamePrice } from '../../src/game-price-tracker-be/domain/entity/game_price';

describe('GamePrice Entity', () => {
  it('should accept a valid Currency object', () => {
    const validCurrency = new Currency(100, 'USD');
    const gamePrice = new GamePrice('Bitcoin', validCurrency);
    expect(gamePrice.name).toBe('Bitcoin');
    expect(gamePrice.currency.amount).toBe(100);
  });
});

File: game-price-tracker-be/core/src/test/kotlin/com/gamepricetracker/core/domain/models/GamePriceTest.kt
package com.gamepricetracker.core.domain.models

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class GamePriceTest {

    @Test
    fun `accepts a valid Currency object`() {
        val validCurrency = Currency(amount = 100.0, code = "USD")
        val gamePrice = GamePrice(name = "Bitcoin", currency = validCurrency)

        gamePrice.name shouldBe "Bitcoin"
        gamePrice.currency.amount shouldBe 100.0
    }
}
